web/views/api/helpers.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Castle.API.Helpers do
+
+  def meta_json(json, %{interval: interval, meta: meta}) do
+    json |> Map.merge(%{interval: interval, meta: meta})
+  end
+  def meta_json(json, %{meta: meta}) do
+    json |> Map.put(:meta, meta)
+  end
+
+  def counts_json(json, %{downloads: downloads}) do
+    json |> Map.put(:downloads, Enum.map(downloads, &count_json/1))
+  end
+  def counts_json(json, %{impressions: impressions}) do
+    json |> Map.put(:impressions, Enum.map(impressions, &count_json/1))
+  end
+
+  defp count_json(data), do: [data.time, data.count]
+
+  def groups_json(json, %{downloads: downloads}) do
+    json |> group_names(downloads) |> Map.put(:downloads, grouped(downloads))
+  end
+  def groups_json(json, %{impressions: impressions}) do
+    json |> group_names(impressions) |> Map.put(:impressions, grouped(impressions))
+  end
+
+  defp group_names(json, data) do
+    names = group_ranks(data) |> Enum.map(&group_find_rank(data, &1))
+    json |> Map.put(:groups, names)
+  end
+
+  defp group_ranks(data) do
+    data |> Enum.map(&(&1.rank)) |> Enum.uniq() |> Enum.sort()
+  end
+
+  defp group_find_rank(data, rank) do
```

Comment:
Re-reading this...

Yes, group_ranks pulls out the ranks and sorts them (it'll end up something like `[1, 2, 3, 4, nil]`).  Then group_find_rank translates those to human-readable names (based on `display`), or "Other" if rank was nil.